ckcp/openshift_dev_setup.sh

```diff
@@ -70,7 +70,19 @@ parse_args() {
   done
 }
 
+# Checks if a binary is present on the local system
+precheck_binary() {
+  for binary in "$@"; do
+    command -v "$binary" >/dev/null 2>&1 || {
+      echo >&2 "openshift_dev_setup.sh requires '$binary' command-line utility to be installed on your local machine. Aborting..."
+      exit 1
+    }
+  done
+}
+
 init() {
+  precheck_binary "kubectl" "yq" "curl"
```

Comment:
I would suggest to add `argodcd CLI` as well to this list - https://argo-cd.readthedocs.io/en/stable/cli_installation/. We're using it in this script for e.g.- to register clusters to Argo CD.

Reply:
Also, would it be a good idea to move this function call to `main()`  since it's checking for binaries? init() is initializing different configs and variables for setting up ckcp and so `precheck_binary` could be called before init(). wdyt?

Reply:
yes makes sense, thx for the feedback :D :D

Reply:
I've just pushed again, PTAL and let me know if it's better now ;)

Reply:
Thanks